src/main/java/Storage.java

```diff
@@ -0,0 +1,110 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with storing and retrieving tasks from a saved file in the hard drive.
+ */
+public class Storage {
+    private final File file;
+
+    /**
+     * Constructor to create a saved file in the hard drive if it does not already exist.
+     * @param path The path where the saved file is located.
+     * @throws IOException If error occur when creating the new file.
+     * @throws DukeException If error occur when creating the folder/file.
+     */
+    public Storage(String path) throws IOException, DukeException {
+        String[] splits = path.split("/");
+        File dir = new File(splits[0]);
+        if (!dir.exists()) {
+            if (!dir.mkdir()) {
+                throw new DukeException("Error while creating ./data folder.");
+            }
+        }
+
+        File savedTasks = new File(path);
+        if (!savedTasks.exists()) {
+            if (!savedTasks.createNewFile()) {
+                throw new DukeException("Error while creating savedTask.txt file.");
+            }
+        }
+
+        this.file = savedTasks;
```

Comment:
Perhaps the `this` can be removed since file is not being shadowed. Similarly for other places.